scripts/import_dvv.sql

```diff
@@ -1,3 +1,131 @@
+-- Update functions --
+
+-- Update loppupvm for previous eldest in the appartment when a new eldest is inserterd.
+--TODO! This should work if we only want a single active eldest per appartment.
+create or replace function update_loppupvm() returns trigger as $$
+begin
+    update jkr.rakennuksen_vanhimmat
+    set loppupvm = new.alkupvm - interval '1 day'
+    where rakennus_id = new.rakennus_id
+    and osapuoli_id != new.osapuoli_id
+    and osapuoli_id != null
+    and huoneistokirjain = new.huoneistokirjain
+    and huoneistonumero = new.huoneistonumero
+    and loppupvm is null;
+    return new;
+end;
+$$ language plpgsql;
+
+-- Update omistuksen_loppupvm for previous owner, when new owner is inserted.
+-- This will set loppupvm to all previous owners of the building, even if only one changed...
+create or replace function update_omistuksen_loppupvm() returns trigger as $$
+begin
+    update jkr.rakennuksen_omistajat
+    set omistuksen_loppupvm = new.omistuksen_alkupvm - interval '1 day'
+    where rakennus_id = new.rakennus_id
+    and osapuoli_id != new.osapuoli_id
+    and osapuoli_id != null
+    and omistuksen_loppupvm is null;
+    return new;
+end;
+$$ language plpgsql;
+
+-- Matches and updates henkilotunnus for osapuoli with missing henkilotunnus. 
+create or replace function update_osapuoli_without_henkilotunnus() returns void as $$
```

Comment:
Voi olla että tämä funktio pitää jättää käyttämättä, koska on periaatteessa mahdollista, ettei kyseessä ole sama henkilö. On turvallisempaa luoda aina henkilötunnukselliset henkilöt uusina osapuolina, ja pitää henkilötunnuksettomat henkilötunnuksettomina, jottei henkilötunnukselle jää merkintää omistuksesta, joka on ollut ilman henkilötunnusta.

Ei siis haittaa, vaikka kohteen omistajissa olisi useita samannimisiä osapuolia eri aikoina. Näin käy joka tapauksessa hyvin usein, koska omistajien nimiä/osoitteita kirjoitetaan eri tavalla. Käytännössä kyseessä on ulkomainen omistaja.